xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -2708,6 +2708,15 @@ bool CActiveAE::SupportsRaw(AEAudioFormat &format)
   return true;
 }
 
+bool CActiveAE::SupportsCoreExtraction(int profile)
+{
+  if (profile != FF_PROFILE_DTS_HD_HRA && profile != FF_PROFILE_DTS_HD_MA)
+    return true;
+
+  return CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+      CSettings::SETTING_AUDIOOUTPUT_DTSHDCOREFALLBACK);
```

Comment:
Please save this into a m_settings member, e.g. allowdtscorepassthrough - that way you don't have to bother the Settings all the time it's called.

AE will pick it up automatically even during playback with:
```
void CActiveAE::OnSettingsChange()
{
  m_controlPort.SendOutMessage(CActiveAEControlProtocol::RECONFIGURE);
}
```